src/v/storage/disk_log_impl.cc

```diff
@@ -349,6 +349,28 @@ ss::future<> disk_log_impl::do_compact(compaction_config cfg) {
         }
 
         auto segment = *seg_it;
+        // since we don't support partially-compacted segments, a segment must
+        // end before the max compactible offset to be eligible for compaction.
+        if (segment->offsets().stable_offset > cfg.max_collectible_offset) {
+            vlog(
+              gclog.debug,
+              "[{}] segment {} stable offs {} > max compactible offs {}, "
+              "skipping.",
+              config().ntp(),
+              segment->reader().filename(),
+              segment->offsets().stable_offset,
+              cfg.max_collectible_offset);
+            continue;
```

Comment:
This needs to be fixed. The surrounding code was rewritten and this first attempt at merging is incorrect--the `continue` here will be an infinite loop. Will update shortly.

Reply:
Done.